lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -130,11 +133,11 @@ def run!(verify_result: true, allow_protected_ns: false, prune: true)
       @logger.phase_heading("Initializing deploy")
       validate_configuration(allow_protected_ns: allow_protected_ns, prune: prune)
       resources = discover_resources
+      resources += secrets_from_ejson
```

Comment:
I dislike that `discover_resources` is now _not_ discovering all resources and we need to make this additional call. Would the abstraction not be more fitting to generate the EJSON secrets inside `discover_resources` itself? This would require updating `TemplateDiscovery` as well as `KubernetesResource::build`. IMO deploy_task shouldn't be in the business of being aware of secrets

Reply:
I'll move the secret appending into `discover_resources` to encapsulate all discovery there, but I'll hold off on touching `TemplateDiscovery` for now unless you feel strongly - processing secrets as templates will probably need to be addressed to resolve [this discussion](https://github.com/Shopify/kubernetes-deploy/pull/235#discussion_r161610047), and I don't want this PR to get any bigger if possible. For now this PR does ~one thing: generalise how secrets are deployed, we can change how they're built in a follow up.